### PR TITLE
Add Kernel API for WebSocket message broadcast

### PR DESCRIPTION
**Pull request by Zuoqiu-Yingyi**
* [x] Please commit to the dev branch
* [ ] For contributing new features, please supplement and improve the corresponding user guide documents
* [ ] For bug fixes, please describe the problem and solution via code comments
* [ ] For text improvements (such as typos and wording adjustments), please submit directly

Add APIs for message broadcast

**Tested** on Windows 11

- `/ws/broadcast`: WebSocket
  - Push broadcast messages
  - Receive broadcast messages
  - **query params**
    - `channel`: broadcast channel name
  - **example**
    - `ws://127.0.0.1:6806/ws/broadcast?channel=your-channel-name`
- `/api/broadcast/channels`: GET request
  - Get all channels' info
  - **response body**
    - ```json
      {
          "code": 0,
          "msg": "",
          "data": {
              "channels": [
                  {
                      "name": "test-channel",
                      "count": 1
                  }
              ]
          }
      }
      ```
- `/api/broadcast/postMessage`: POST request
  - Push broadcast messages
  - **request body**
    - ```json
      {
          "channel": "your-channel-name",
          "message": "your-message-content"
      }
      ```
      - `channel`: the name of channel
      - `message`: the message will be broadcast
  - **response body**
    - ```json
      {
          "code": 0,
          "msg": "",
          "data": {
              "channel": {
                  "name": "your-channel-name",
                  "count": 1
              }
          }
      }
      ```
      - `data.channel.count`: the count of channel listener
    - ```json
      {
          "code": -1,
          "msg": "broadcast channel [your-channel-name] not found",
          "data": null
      }
      ```
    - ```json
      {
          "code": -2,
          "msg": "broadcast message failed: ...",
          "data": null
      }
      ```
- `/api/broadcast/getChannelInfo`: POST request
  - get the info of a channel
  - **request body**
    - ```json
      {
          "name": "your-channel-name"
      }
      ```
      - `name`: the name of channel
  - **response body**
    - ```json
      {
          "code": 0,
          "msg": "",
          "data": {
              "channel": {
                  "name": "your-channel-name",
                  "count": 1
              }
          }
      }
      ```
      - `data.channel.name`: the name of channel
      - `data.channel.count`: the count of channel listener
    - ```json
      {
          "code": -1,
          "msg": "broadcast channel [your-channel-name] not found",
          "data": null
      }

### CODE REVIEW

File: kernel/api/broadcast.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+@returns
+
+	body.data.channel: the channel information
+*/
+func getChannelInfo(c *gin.Context) {
+	ret := gulu.Ret.NewResult()
+	defer c.JSON(http.StatusOK, ret)
+
+	arg, ok := util.JsonArg(c, ret)
+	if !ok {
```
Comment (zuoez02) [inline kernel/api/broadcast.go:174]: 无参数应该返回400而不是200 OK吧

File: kernel/api/router.go [hunk 2]
```diff
@@ -372,4 +372,9 @@ func ServeAPI(ginServer *gin.Engine) {
 	ginServer.Handle("POST", "/api/petal/setPetalEnabled", model.CheckAuth, model.CheckReadonly, setPetalEnabled)
 
 	ginServer.Handle("POST", "/api/network/forwardProxy", model.CheckAuth, model.CheckReadonly, forwardProxy)
+
+	ginServer.Handle("GET", "/ws/broadcast", model.CheckAuth, broadcast)
+	ginServer.Handle("GET", "/api/broadcast/channels", model.CheckAuth, getChannels)
```
Comment (zuoez02) [inline kernel/api/router.go:377]: 按思源的搞法，基本都是post，除非弄文件这种需要用Get请求

File: kernel/api/broadcast.go (hunk 1 above)
Comment (Zuoqiu-Yingyi) [reply kernel/api/broadcast.go:174]: 我看其他地方都是用的自定义的响应码

File: kernel/api/router.go (hunk 2 above)
Comment (Zuoqiu-Yingyi) [reply kernel/api/router.go:377]: 没有参数的请求用 GET 多好~

**Pull request merged**